### ISSUE TITLE
Suppressing `UnusedClass` for all service classes in the container bypasses dead-code detection

### ISSUE THREAD

**Issue by tunterreitmeier** [psalm/psalm-plugin-symfony#381]
See also #380 

Assuming a pretty standard container configuration like this:

```php
return App::config([
    'services' => [
        'App\\' => [
            'resource' => '../src/',
            'exclude' => '../src/{SomeDirectory,AnotherDirectory,Kernel.php}',
        ],
    ],
]);
````

Basically every class in `src/` will show up as a service in the debug container. Suppressing `UnusedClass` is a great quick fix to suppress quite a few false positives that pop up, because Psalm cannot statically infer the usage. Also, because Psalm does not see any class usage, it does not even check method calls. So status quo is that quite a few cases of dead code are not detected.

Examples are:
- autoconfigured aliases of interfaces with a single implementation (you declare an interface as the dependency and container dynamically instantiates the implementation) 
- Controller
- EventListener / Subscriber
- Messenger MessageHandler
... and probably quite a few more. Basically everything that is called by a Symfony component instead of your apllication code.

I would like to work on a solution but also wanted to discuss potential solutions first. Seems like there is no real easy solution but it's more of a "teach Psalm how Symfony works" thing.

Happy for any ideas and feedback. Thanks!

**Comment by tunterreitmeier** [comment]
Update: this is a [demo repo ](https://github.com/tunterreitmeier/symfony-psalm-dead-code)with quite a bit of dead code to showcase this. Added a [action](https://github.com/tunterreitmeier/symfony-psalm-dead-code/actions/workflows/ci.yml) that reports no errors.
Already started fiddling around with solutions: https://github.com/tunterreitmeier/psalm-plugin-symfony/tree/fix/mark-symfony-usages
Appraoch is basically to add the appropriate callers, where Psalm does not statically see any usage. `UnusedClass` is more of a symptom, that Psalm does not understand the call structure as is.

So far, the only issue are routes. To get all routes, you would have to boot a Kernel, which would open up a whole can of worms. Best solution would probably be to read the `url_generating_routes.php` cache, which is probably different per Symfony version.
So at the moment, I am only checking the `#[Route]` attribute, which does not find e.g. routes from a config file.
